src/base_classes/mo_space_info.cc

```diff
@@ -278,11 +298,50 @@ std::pair<SpaceInfo, bool> MOSpaceInfo::read_mo_space(const std::string& space,
     return std::make_pair(space_info, read);
 }
 
+std::pair<SpaceInfo, bool>
+MOSpaceInfo::read_mo_space_from_map(const std::string& space,
+                                    std::map<std::string, std::vector<size_t>>& mo_space_map) {
+    bool read = false;
+    psi::Dimension space_dim(nirrep_);
+    std::vector<MOInfo> vec_mo_info;
+
+    // lookup the space
+    auto it = mo_space_map.find(space);
+    if (it != mo_space_map.end()) {
+        const auto& dim = mo_space_map[space];
+        if (dim.size() == nirrep_) {
+            for (size_t h = 0; h < nirrep_; ++h) {
+                space_dim[h] = dim[h];
+            }
+            read = true;
+        } else {
+            outfile->Printf("\n  The size of space \"%s\" (%d) does not match the number of "
```

Comment:
It looks like it would be a good idea to throw an exception here. For example 
```suggestion
            throw std::runtime_error("The size of space \"%s\" (%d) does not match the number of "...
```